### ISSUE TITLE
CORS issue : Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### ISSUE THREAD

**Issue by malwaremanu** [tiangolo/fastapi#5453]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
origins = ["https://abcd-1234.preview.abc.com",
    "https://something.netlify.app","http://localhost:3000/",
    "http://localhost.tiangolo.com","https://localhost.tiangolo.com"
    ]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

### Description

Facing issue with : "https://abcd-1234.preview.abc.com

Access to XMLHttpRequest at 'https://server.com/login' from origin 'https://abcd-1234.preview.abc.com' has been blocked by CORS policy: **Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.**

Whereas "https://something.netlify.app" is working fine.
Please suggest what to do ?

### Operating System

Linux

### Operating System Details

I am using [codesandbox](https://codesandbox.io/) for frontend application and deta.sh for backend fastapi.

### FastAPI Version

0.83.0

### Python Version

Python 3.9

**Comment by novitae** [comment]
hey, I currently have the same issue, the cors headers aren't responded. I can use the copy pasted script from the documentation (https://fastapi.tiangolo.com/tutorial/cors/), it won't work
<img width="1437" alt="Capture d’écran 2022-11-01 à 11 51 19" src="https://user-images.githubusercontent.com/85891169/199217300-af770ef9-ef14-4534-8996-608614647250.png">

**Comment by tiangolo** [comment]
Thanks for coming back to close the issue @malwaremanu ! ☕

@novitae it's probably better if you create a new issue, following the template, with a full self contained example, etc.

**Comment by malwaremanu** [comment]
> hey, I currently have the same issue, the cors headers aren't responded. I can use the copy pasted script from the documentation (https://fastapi.tiangolo.com/tutorial/cors/), it won't work <img alt="Capture d’écran 2022-11-01 à 11 51 19" width="1437" src="https://user-images.githubusercontent.com/85891169/199217300-af770ef9-ef14-4534-8996-608614647250.png">

Can you share your code for "app.add_middleware" ?
